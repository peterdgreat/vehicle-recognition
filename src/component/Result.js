import React from 'react';
import { useSelector } from 'react-redux';
import {NavLink} from 'react-router-dom';
import '../styles/Result.css';

export default function Result() {
      const dataResult =useSelector((state) => state.vehicleReducer.vehicles);
    return (
       <div>
             <NavLink to='/'>
        <button>Back</button>
</NavLink>

            {
              dataResult !==undefined && (
                                     <div>
                                      { dataResult[0].car.make} {dataResult[0].car.model} {dataResult[0].car.years} {dataResult[0].color.name}
                                    </div>    
                                    )
            }
            
       </div>
    )
}


