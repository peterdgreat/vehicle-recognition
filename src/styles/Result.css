.result {
    color: #fff;
    width: 80vw;
    height: 80vh;
}

.card {
    background-color: transparent;
}

.list-group-item {
    background-color: transparent;
    color: #fff;
}