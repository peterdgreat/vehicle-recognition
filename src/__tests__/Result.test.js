import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Result from '../component/Result';
import store from '../redux/configurestore';

describe('Test Result To Match Snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
       <Router>
        <Routes>
           <Route path="/" element={<Result />} />
        </Routes>
         </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Result', () => {
  test('Expect Test To be Truthy ', () => {
    render(
      <Provider store={store}>
       <Router>
        <Routes>
           <Route path="/" element={<Result />} />
        </Routes>
         </Router>
      </Provider>,
    );
    const header = screen.findByText('No Car Uploaded');
    expect(header).toBeTruthy();
  });
});

