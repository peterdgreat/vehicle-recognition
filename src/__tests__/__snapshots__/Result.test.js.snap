// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Result To Match Snapshot renders correctly 1`] = `
<div>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <button
      className="btn"
    >
      Back
    </button>
  </a>
  <div
    className="result card border"
  >
    <h1
      className="h-100 d-flex justify-content-center align-items-center"
    >
      No Car Uploaded
    </h1>
  </div>
</div>
`;
